hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/parse/QueryConverter.java

```diff
@@ -156,32 +161,28 @@ private HazelcastSqlToRelConverter createSqlToRelConverter() {
      *  It is used instead of "expand" flag due to bugs in Calcite (see {@link #EXPAND}).
      * </li>
      * <li>
-     *  Transformation of distinct UNION to UNION ALL, merging the neighboring UNION relations.
+     *  Check, if the relation uses cyclic user types, and if they are allowed - skip this step.
      * </li>
      * <li>
-     *  Check, if the relation uses cyclic user types, and if they are allowed - skip this step.
+     *  Extract unsupported source expressions from an UPDATE stmt into a {@link Calc}.
```

Comment:
you moved ExtractUpdateExpressionsRule to latter phase

Reply:
https://github.com/hazelcast/hazelcast/pull/26032/commits/a8179a6e5e9e533507a41e020d5f7bf0a5d85b47